app/models/assignment.rb

```diff
@@ -53,15 +56,20 @@ def started_six_months_ago?
   end
 
   scope :active, (-> { not_ended.started })
-  scope :inactive, (-> { ended })
+  scope :inactive, (-> { ended.or(no_start) })
 
   scope :active_between, lambda { |start_date, end_date|
-    start_within(start_date..end_date)
-      .or(end_within(start_date..end_date))
-      .or(no_end.start_before(start_date))
+    no_end.start_before(end_date)
+          .or(
```

Comment:
Could it also need `no_start.end_after(start_date)`?

Reply:
No, because a assignment with no start is not, and was never active